src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/IChannelExecutorPool.cs

```diff
@@ -0,0 +1,51 @@
+namespace MassTransit.EventHubIntegration
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Util;
+
+
+    public interface IChannelExecutorPool
+    {
+        Task Push(string partition, Func<Task> method, CancellationToken cancellationToken = default);
+    }
+
+
+    public class ChannelExecutorPool :
```

Comment:
I do not think `ConcurrentDictionary` suits better here, as in most cases it is created and filled will all data during client initialisation, at least for Kafka and EventHub. Partition initialisation event will create executor per partition. Then it just access them for processing. in thin implementation Get without lock (and it will be the main method called)

Reply:
Oh, right, I forgot it's only locked on mutate.